### PR TITLE
Week3

### PR DESCRIPTION
**Pull request by joudi6**

### CODE REVIEW

File: homework-submission/todo-app/src/App.js [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+  }
+
+  handleCompleted = index => {
+    const todos = [...this.state.todos];
+    todos[index].done = !todos[index].done;
+    this.setState({ todos });
+  };
+
+  handleSubmit = newItem => {
+    const oldState = this.state.todos;
```
Comment (jalal75) [inline homework-submission/todo-app/src/App.js]: const oldState = [...this.state.todos]; 👍

File: homework-submission/todo-app/src/App.js [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+    const ItemData = oldState.push(newItem);
+    this.setState({ ItemData });
```
Comment (jalal75) [inline homework-submission/todo-app/src/App.js]: this.setState({todos:  ItemData });

File: homework-submission/todo-app/src/App.js [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+  render() {
+    const data = this.state.todos.map((todo, key) => (
+      <Item
+        todo={todo}
+        key={key}
+        index={key}
+        onCompleted={this.handleCompleted}
+        onEdit={this.handleEdit}
+        onRemove={this.handleRemove}
+        onCancel={this.handleCancel}
```
Comment (jalal75) [inline homework-submission/todo-app/src/App.js]: not needed 
??????

File: homework-submission/todo-app/src/components/MainForm.js [hunk 4]
```diff
@@ -0,0 +1,28 @@
+import React, { Component } from "react";
+const randomize = require("randomatic");
+
```
Comment (jalal75) [inline homework-submission/todo-app/src/components/MainForm.js]: import randomize from "randomatic"

File: homework-submission/todo-app/src/App.js (hunk 3 above)
Comment (joudi6) [reply homework-submission/todo-app/src/App.js]: it gives error if commented out (TypeError: _this.props.onUpdate is not a function) in Item.js:9

Comment (jalal75) [reply homework-submission/todo-app/src/App.js]: i am talking about : 
  onCancel={this.handleCancel}

**Pull request closed without merging**